db_test.go

```diff
@@ -241,6 +241,60 @@ func TestModify(t *testing.T) {
 	assert.Equal(t, "44", (user2).(map[string]interface{})["age"])
 }
 
+func TestSmartQuery(t *testing.T) {
```

Comment:
These tests are really good, thanks for thinking of them! They can be better separated into sub tests though, so that it is known which properties they are testing (I am finding it hard to track state and reasoning, but I think I see what you are doing).

The way you can do that is with
```
func TestSmartQuery(t *testing.T) {
   someSetup()
   t.Run("can read select queries", func(t *testing.T) {
       testCodeHere()
   })
   t.Run("can read delete queries with nil", func (t *testing.T) {
       moreTestCodeHere()
   })
}
```
This still shares setup and order of execution, but it gives a clear separation into which properties of the code are being tested. Tracking state is also easier, since the subtests should only rely on the shared state (even though the ordered state affects them).

Reply:
Done.